enhancements/proposals/2490-scylladbmonitoring-external-prometheus/README.md

```diff
@@ -0,0 +1,611 @@
+       // +kubebuilder:default:="prometheus"
+       Name string `json:"name,omitempty"`
+
+       // type is the type of the datasource. Only "prometheus" is supported.
+       // +kubebuilder:default:="Prometheus"
+       // +optional
+       Type GrafanaDatasourceType `json:"type,omitempty"`
+
+       // url is the URL of the datasource.
+       // +kubebuilder:validation:MinLength=1
+       URL string `json:"url"`
+
+       // prometheus defines Prometheus-specific options.
+       // +optional
+       PrometheusOptions *GrafanaPrometheusDatasourceOptions `json:"prometheusOptions,omitempty"`
+   }
+   ```
+
+   Strongly-typed enum will be used for the datasource type:
+
+   ```go
+   // GrafanaDatasourceType defines the type of Grafana datasource.
+   // +kubebuilder:validation:Enum="Prometheus"
+   type GrafanaDatasourceType string
+
+   const (
+       // GrafanaDatasourceTypePrometheus is the Prometheus datasource type.
+       GrafanaDatasourceTypePrometheus GrafanaDatasourceType = "Prometheus"
+   )
+   ```
+
+   `GrafanaPrometheusDatasourceSpec` will define Prometheus-specific options, including TLS and authentication.
+
+   ```go
+   type GrafanaPrometheusDatasourceOptions struct {
+      // tls holds TLS configuration for connecting to Prometheus over HTTPS.
+      // +optional
+      TLS *GrafanaDatasourceTLSSpec `json:"tls,omitempty"`
+
+      // auth holds authentication options for connecting to Prometheus.
+      // +optional
+      Auth *GrafanaPrometheusDatasourceAuthSpec `json:"auth,omitempty"`
+   }
+   ```
+
+   `GrafanaDatasourceTLSSpec` will define TLS configuration options for server certificate verification and client
+   authentication.
+
+   ```go
+   // LocalObjectKeySelector selects a key of a ConfigMap or Secret in the same namespace.
+   type LocalObjectKeySelector struct {
+      // name of the selected object.
+      // +kubebuilder:validation:MinLength=1
+      Name string `json:"name"`
+
+      // key within the selected object.
+      // +kubebuilder:validation:MinLength=1
+      Key string `json:"key"`
+   }
+
+   type GrafanaDatasourceTLSSpec struct {
+      // caCert is a reference to a key within the CA bundle ConfigMap. The key should hold the CA cert in PEM format.
+      // When not specified, system CAs are used.
+      // +optional
+      CACertConfigMapRef *LocalObjectKeySelector `json:"caCertConfigMapRef,omitempty"`
+
+      // insecureSkipVerify controls whether to skip server certificate verification.
+      // +kubebuilder:default:=false
+      // +optional
+      InsecureSkipVerify bool `json:"insecureSkipVerify,omitempty"`
+
+	  // clientTLSKeyPairSecretRef is a name of a Secret holding client TLS certificate and key for mTLS authentication.
+	  // It's expected to be a standard Kubernetes TLS Secret with `tls.crt` and `tls.key` keys.
+      // +optional
+      ClientTLSKeyPairSecretName *corev1.LocalObjectReference `json:"clientTLSKeyPairSecretRef,omitempty"`
```

Comment:
https://github.com/scylladb/scylla-operator/blob/9f1d676c34804a6bd794111b107ef6eb57f0c774/vendor/k8s.io/api/core/v1/types.go#L7090-L7099

Reply:
Switched to a local type.